prow/flagutil/github.go

```diff
@@ -57,9 +57,9 @@ func (o *GitHubOptions) AddFlags(fs *flag.FlagSet) {
 // Validate validates GitHub options.
 func (o *GitHubOptions) Validate(dryRun bool) error {
 	endpoints := o.endpoint.Strings()
-	for _, uri := range endpoints {
+	for i, uri := range endpoints {
 		if uri == "" {
-			uri = github.DefaultAPIEndpoint
+			endpoints[i] = github.DefaultAPIEndpoint
```

Comment:
can you add a test that demonstrates the bug that this is fixing?

Reply:
Added a number of tests for the Validate function. Cleaned up Validate slightly as some of the code there is unreachable. Hope it is all okay.

Reply:
The bug here is that the range loop over endpoints takes a copy of each slice element. The copy of the slice element ```uri``` is set to the default API endpoint which has absolutely no effect.

I think that the expected behaviour here should be that if any of the string elements in the endpoints slice is empty its value should be defaulted to the default API endpoint.